src/services/inlayHints.ts

```diff
@@ -267,6 +271,14 @@ namespace ts.InlayHints {
             if (!signature) {
                 return;
             }
+            const shouldWrap = shouldWrapParameterList(node);

```

Comment:
There's another subtlety to this - I think you should only wrap a parameter list if the user's settings indicate that a return annotation will be added.

Hoist the check for whether or not a return annotation should be added above, and factor it into `shouldWrap`.

Also, rename this to `parameterNeedsParens`

Reply:
Thanks!  I've extended a new parameter `autoWrap` to control whether or not to wrap the parameter automatically.

Reply:
Also, I posted two screenshot on the description for the effect